File: src/main/java/br/com/repository/Usuario/UsuarioRepository.java
package br.com.repository.usuario;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import br.com.model.entities.classes.usuario.Usuario;

@Repository
public interface UsuarioRepository extends CrudRepository<Usuario, Integer> {
    public Usuario findByEmail(String email);

    public Usuario findByCnpjCpf(String cnpjCpf);
}

File: src/main/java/br/com/repository/usuario/UsuarioRepository.kt
package br.com.repository.usuario

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import br.com.model.usuario.Usuario
import jakarta.transaction.Transactional

@Repository
interface UsuarioRepository : JpaRepository<Usuario, Int> {
    @Transactional
    fun findByEmail(email: String): Usuario
    @Transactional
    fun findByCnpjCpf(cnpjCpf: String): Usuario
}
